resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/minikube/Minikube.scala

```diff
@@ -16,100 +16,73 @@
-    outputs.head
-  }
+  lazy val minikubeVersionString =
+    executeMinikube("version").find(_.contains(VERSION_PREFIX)).get
 
-  def getMinikubeStatus: MinikubeStatus.Value = {
-    val statusString = executeMinikube("status")
-    logInfo(s"Minikube status command output:\n$statusString")
-    // up to minikube version v0.30.0 use this to check for minikube status
-    val oldMinikube = statusString
-      .filter(line => line.contains(MINIKUBE_VM_PREFIX) || line.contains(MINIKUBE_PREFIX))
-
-    if (oldMinikube.isEmpty) {
-      getIfNewMinikubeStatus(statusString)
-    } else {
-      val finalStatusString = oldMinikube
-        .head
-        .replaceFirst(MINIKUBE_VM_PREFIX, "")
-        .replaceFirst(MINIKUBE_PREFIX, "")
-      MinikubeStatus.unapply(finalStatusString)
-        .getOrElse(throw new IllegalStateException(s"Unknown status $statusString"))
-    }
-  }
+  def logVersion(): Unit =
+    logInfo(minikubeVersionString)
 
   def getKubernetesClient: DefaultKubernetesClient = {
-    val kubernetesMaster = s"https://${getMinikubeIp}:8443"
-    val userHome = System.getProperty("user.home")
-    val minikubeBasePath = Paths.get(userHome, MINIKUBE_PATH).toString
-    val profileDir = if (Files.exists(Paths.get(minikubeBasePath, "apiserver.crt"))) {
-      // For Minikube <1.9
-      ""
-    } else {
-      // For Minikube >=1.9
-      Paths.get("profiles", executeMinikube("profile")(0)).toString
+    // only the three-part version number is matched (the optional suffix like "-beta.0" is dropped)
+    val versionArrayOpt = "\\d+\\.\\d+\\.\\d+".r
+      .findFirstIn(minikubeVersionString.split(VERSION_PREFIX)(1))
+      .map(_.split('.').map(_.toInt))
+
+    versionArrayOpt match {
+      case Some(Array(x, y, z)) =>
+        if (Ordering.Tuple3[Int, Int, Int].lt((x, y, z), (1, 7, 3))) {
+          assert(false, s"Unsupported Minikube version is detected: $minikubeVersionString." +
+            "For integration testing Minikube version 1.7.3 or greater is expected.")
+        }
+      case _ =>
+        assert(false, s"Unexpected version format detected in `$minikubeVersionString`." +
+          "For minikube version a three-part version number is expected (the optional " +
+          "non-numeric suffix is intentionally dropped)")
     }
-    val apiServerCertPath = Paths.get(minikubeBasePath, profileDir, "apiserver.crt")
-    val apiServerKeyPath = Paths.get(minikubeBasePath, profileDir, "apiserver.key")
-    val kubernetesConf = new ConfigBuilder()
-      .withApiVersion("v1")
-      .withMasterUrl(kubernetesMaster)
-      .withCaCertFile(
-        Paths.get(userHome, MINIKUBE_PATH, "ca.crt").toFile.getAbsolutePath)
-      .withClientCertFile(apiServerCertPath.toFile.getAbsolutePath)
-      .withClientKeyFile(apiServerKeyPath.toFile.getAbsolutePath)
-      .build()
-    new DefaultKubernetesClient(kubernetesConf)
+
+    new DefaultKubernetesClient(Config.autoConfigure("minikube"))
```

Comment:
If we're now enforcing that the name of the context must be `minikube` this should be included in the apache/spark-website#321 PR

Reply:
I think that is the default context for minikube. Although I have not found this in documentation so I asked this in the Minikube user's list:

```
Hi!

As I see the default context for the started cluster is `minikube` but I have not found this in documentation.
- Am I right? 
- Can I really on this?
- Could you please show me the documentation about this (or the relevant code part)?

Thanks in advance,
Attila
```

I hope we can get an answer soon.

Reply:
I'm sure it's probably is the default (though their community will be best placed to confirm).  The scenario I was thinking of was more the one where users have renamed their contexts to their own preference.  I have access to a lot of R&D clusters in my $dayjob and at times I've had as many as 10+ different contexts in my `KUBECONFIG` so I would rename contexts appropriately as some tools (GKE I'm looking at you) produce rather user unfriendly names by default.  Not sure I've ever renamed the `minikube` context itself but it's a thing that users can and might do.

TL;DR If there's a naming assumption present in the code, particularly where it pertains to user managed configuration files outside of Spark's control, it should be documented as such.

Reply:
I have tested this case and when after minikube is started and the context is switched to another one even that makes the test fail. 

So the best probably is mentioning that fact too:
```
- the current kubernetes context must be minikube's default context (called 'minikube'). This can be selected by `minikube kubectl -- config use-context minikube`. This is only needed when after minikube is started another kubernetes context is selected.
```